qutebrowser/mainwindow/tabwidget.py

```diff
@@ -603,19 +608,17 @@ def tabSizeHint(self, index: int):
                 width = main_window.width() * perc / 100
             else:
                 width = int(confwidth)
-            size = QSize(max(minimum_size.width(), width), height)
+            size = QSize(width, height)
         else:
             if config.cache['tabs.pinned.shrink'] and self._tab_pinned(index):
                 # Give pinned tabs the minimum size they need to display their
                 # titles, let Qt handle scaling it down if we get too small.
                 width = self.minimumTabSizeHint(index, ellipsis=False).width()
             else:
                 # Request as much space as possible so we fill the tabbar, let
-                # Qt shrink us down
-                width = self.width()
-
-            # If we don't have enough space, we return the minimum size
-            width = max(width, minimum_size.width())
+                # Qt shrink us down. If for some reason (tests, bugs)
+                # self.width() gives 0, use a sane min of 10 px
+                width = max(self.width(), 10)
```

Comment:
It should be safe to give a width larger than the bar size, this is mostly just for weird situations (for some reason, in tests, `self.width()` is 0, which causes parts later on to fail since we recommend a tab of size zero here).

I'm not sure if there's a better way of handling this, but this seems reasonable to me.